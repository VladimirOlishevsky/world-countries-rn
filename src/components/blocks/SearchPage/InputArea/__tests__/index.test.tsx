import React from 'react';
import { InputArea } from '..';
import { render, fireEvent } from '@testing-library/react-native';

const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: mockedNavigate,
      goBack: mockedNavigate
    }),
  };
});

jest.mock('react-native-textinput-effects');
// jest.mock('react-native-vector-icons/FontAwesome', () => ({ Icon: '' }));

describe('InputArea', () => {

  beforeEach(() => {
    jest.clearAllMocks();
  });
  test('render', () => {
    const tree = render(<InputArea />);
    expect(tree).toMatchSnapshot()
  });
  // test('onPress navigate', async () => {
  //   const { findByTestId } = render(<BackButton page={page}/>);
  //   const button = await findByTestId('go back')
  //   expect(mockedNavigate).not.toBeCalled();
  //   fireEvent(button, 'press');
  //   expect(mockedNavigate).toHaveBeenCalledTimes(1);
  // });
  // test('onPress go back', async () => {
  //   const { findByTestId } = render(<BackButton />);
  //   const button = await findByTestId('go back')
  //   expect(mockedNavigate).not.toBeCalled();
  //   fireEvent(button, 'press');
  //   expect(mockedNavigate).toHaveBeenCalledTimes(1);
  // });
});
