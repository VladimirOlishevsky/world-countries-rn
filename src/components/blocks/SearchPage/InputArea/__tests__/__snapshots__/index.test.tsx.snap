// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputArea render 1`] = `
<View
  style={
    Object {
      "paddingBottom": NaN,
      "paddingTop": NaN,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#6200ee",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "marginTop": undefined,
        "minWidth": 64,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "marginLeft": 12,
              "marginRight": -4,
            }
          }
        >
          <Text
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "color": "#ffffff",
                  "fontSize": 16,
                },
              ]
            }
          >
            â–¡
          </Text>
        </View>
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Search
        </Text>
      </View>
    </View>
  </View>
</View>
`;
